main.go

```diff
@@ -0,0 +1,209 @@
+	"github.com/jmoiron/sqlx"
+)
+
+var (
+	db *sqlx.DB
+)
+
+func main() {
+	_db, err := sqlx.Connect(
+		"mysql", 
+		fmt.Sprintf("%s:%s@tcp(%s:%s)/%s?charset=utf8&parseTime=True&loc=Local", 
+			os.Getenv("DB_USERNAME"),
+			os.Getenv("DB_PASSWORD"),
+			os.Getenv("DB_HOSTNAME"),
+			os.Getenv("DB_PORT"),
+			os.Getenv("DB_DATABASE")))
+	if err != nil {
+		log.Fatalf("Cannot Connect to Database: %s", err)
+	}
+	db = _db
+
+	store, err := mysqlstore.NewMySQLStoreFromConnection(db.DB, "sessions", "/", 60*60*24, []byte("secret-token"))
+	if err != nil {
+		panic(err)
+	}
+
+	e := echo.New()
+	e.Use(middleware.Logger())
+	e.Use(session.Middleware(store))
+
+	e.GET("/ping", func(c echo.Context) error {
+		return c.String(http.StatusOK, "pong")
+	})
+	e.POST("/signup", postSignUpHandler)
+	e.POST("/login", postLoginHandler)
+
+	withLogin := e.Group("")
+	withLogin.Use(checkLogin)
+	withLogin.GET("/home", getTweetHandler)
+	withLogin.POST("/home", postTweetHandler)
+	withLogin.GET("/:userid", getAccountHome)
+
+	e.Start(":11800")
+}
+
+type LoginRequestBody struct {
+	Username string `json:"username,omitempty" form:"username"`
+	Password string `json:"password,omitempty" form:"password"`
+}
+
+type User struct {
+	Username   string `json:"username,omitempty"  db:"Username"`
+	HashedPass string `json:"-"  db:"HashedPass"`
+}
+
+func postSignUpHandler(c echo.Context) error {
+	req := LoginRequestBody{}
+	err := c.Bind(&req)
+	if err != nil {
+		return c.String(http.StatusInternalServerError, fmt.Sprintf("binding error: %v", err))
+	}
+
+	if req.Password == "" || req.Username == "" {
+		return c.String(http.StatusBadRequest, "項目が空です")
+	}
+
+	hashedPass, err := bcrypt.GenerateFromPassword([]byte(req.Password), bcrypt.DefaultCost)
+	if err != nil {
+		return c.String(http.StatusInternalServerError, fmt.Sprintf("bcrypt generate error: %v", err))
+	}
+
+	var count int
+
+	err = db.Get(&count, "SELECT COUNT(*) FROM users WHERE Username=?", req.Username)
+	if err != nil {
+		return c.String(http.StatusInternalServerError, fmt.Sprintf("db error: %v", err))
+	}
+
+	if count > 0 {
+		return c.String(http.StatusConflict, "ユーザーが既に存在しています")
+	}
+
+	_, err = db.Exec("INSERT INTO users (Username, HashedPass) VALUES (?, ?)", req.Username, hashedPass)
+	if err != nil {
+		return c.String(http.StatusInternalServerError, fmt.Sprintf("db error: %v", err))
+	}
+	return c.NoContent(http.StatusCreated)
+}
+
+func postLoginHandler(c echo.Context) error {
+	req := LoginRequestBody{}
+	err := c.Bind(&req)
+	if err != nil {
+		return c.String(http.StatusInternalServerError, fmt.Sprintf("binding error: %v", err))
+	}
+
+	user := User{}
+	err = db.Get(&user, "SELECT * FROM users WHERE username = ?", req.Username)
+	// users { Username, HashedPass }
+
+	if err != nil {
+		return c.String(http.StatusInternalServerError, fmt.Sprintf("db error: %v", err))
+	}
+
+	err = bcrypt.CompareHashAndPassword([]byte(user.HashedPass), []byte(req.Password))
+	if err != nil {
```

Comment:
```suggestion
	if err = bcrypt.CompareHashAndPassword([]byte(user.HashedPass), []byte(req.Password)); err != nil {
```
変数のスコープが小さくて良い場合にはこういうふうに書くことが推奨されます。

Reply:
`err = `のとこ`err := `ですね